bundles/org.smarthomej.transform.basicprofiles/src/main/java/org/smarthomej/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -0,0 +1,226 @@
+                    String itemState = item.getState().toString();
+
+                    if (!condition.matches(itemState)) {
+                        allConditionsMet = false;
+                    }
+                } catch (ItemNotFoundException e) {
+                    logger.warn(
+                            "Cannot find item '{}' in registry - check your condition expression - skipping state update",
+                            condition.itemName);
+                    allConditionsMet = false;
+                }
+
+            }
+            if (allConditionsMet) {
+                return state;
+            } else {
+                return configMismatchState;
+            }
+        } else {
+            logger.warn(
+                    "No configuration defined for StateFilterProfile (check for log messages when instantiating profile) - skipping state update");
+        }
+
+        return null;
+    }
+
+    @Nullable
+    State parseState(@Nullable String stateString) {
+        // Quoted strings are parsed as StringType
+        if (stateString == null) {
+            return null;
+        } else if (stateString.startsWith("'") && stateString.endsWith("'")) {
+            return new StringType(stateString.substring(1, stateString.length() - 1));
+        } else {
+            return TypeParser.parseState(acceptedDataTypes, stateString);
+        }
+    }
+
+    class Condition {
+        String itemName;
+
+        ComparisonType comparisonType;
+        String value;
+
+        boolean quoted = false;
+
+        public Condition(String itemName, ComparisonType comparisonType, String value) {
+            this.itemName = itemName;
+            this.comparisonType = comparisonType;
+            this.value = value;
+            this.quoted = value.startsWith("'") && value.endsWith("'");
+            if (quoted) {
+                this.value = value.substring(1, value.length() - 1);
+            }
+        }
+
+        public boolean matches(String state) {
+            switch (comparisonType) {
+                case EQ:
+                    if (!state.equals(value)) {
+                        return false;
+                    }
+                    break;
+                case NEQ: {
+                    if (state.equals(value)) {
+                        return false;
+                    }
+                    break;
+                }
+                default:
+                    logger.warn("Unknown condition type {}. Expected 'eq' or 'neq' - skipping state update",
+                            comparisonType);
+                    return false;
+
+            }
+            return true;
+        }
+
+        enum ComparisonType {
+            EQ,
+            NEQ
+        }
+
+        @Override
+        public String toString() {
+            return "Condition{" + "itemName='" + itemName + '\'' + ", comparisonType=" + comparisonType + ", value='"
+                    + value + '\'' + '}';
```

Comment:
```suggestion
            return "Condition{itemName='" + itemName + "', comparisonType=" + comparisonType + ", value='"
                    + value + "'}';
```

looks like the auto-generated `toString` from IntelliJ. I usually adjust that a bit.